### ISSUE TITLE
Carnage on crafts from very old TweakScale versions.

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/KSPe#46]

**Comment by Lisias** [comment]
The `KSPe.Light.TweakScale` last known to **do not** trigger the problem is [2.3.0.4](https://github.com/net-lisias-ksp/KSPe/tree/RELEASE/Light/2.3.0.4/TweakScale). The tag dates Aug 18, 2021. It fits withe the `KSPe.Light` distributed from TS 2.4.5.2 ti TS 2.4.5.9, where the problem is known to **do not happen**.

Compiling it from this tag reproduced a binary with the same size the one distributed on TS 2.4.5.9. However, its' not binary identical to the one distributed at that time!

Worst, running it I got the problem!!

I want to stress that this is happening while running on KSP 1.4.3, what rules out anything relased to using C# 4.7 - this thing runs on C# 3.5.

This appears to be a **COMPILER** problem, not a runtime one.

**Comment by Lisias** [comment]
It's almost without practical use, but whatever - out of curiosity, I redid the last test using the very first `KSPe.Light` ever, the [2.1.0.11 for TweakScale](https://github.com/net-lisias-ksp/KSPe/tree/RELEASE/Light/2.1.0.11/TweakScale).

Same *krap*. Oh, well - at least this ruled out `KSPe.Light` from the equation **interelly**. This very first  release have 10k in size, it's essentially the first version of the `ConfigNodeWithSteroids` stung and a `MessageBox` helper. There's nothing left to cause problems.

So I'm just removing things from the compilation unit to see what happens.

**Comment by Lisias** [comment]
Well, that thing that would not have any practical use ended up **solving** the mystery.

I removed everything from that damned thing until only the `AssemblyInfo` file was left (a 3.5K DLL!) - and the misbehaviour was reproduced the same.

So… Yeah, the source of my misfortune is something on the `AssemblyInfo` file. Interesting, uh?

I spent the whole night playing combinatorial analysis with this pretty minimalistic DLL. Changed everything, from Assemby Name on the project file to every single datum on the `AssemblyInfo` file.

And the problem was happening the same.

So, out of exasperation, I removed the DLL from the TweakScale's directory and… Now the problem is happening all the time, with or without this extra DLL on the TweakScale's directory!!!

So this extra DLL was never needed to trigger the problem, the cause of it is something else and by pure chance I managed to trigger the damned thing by adding, removing or changing the DLL from the Kraken damned directory!

**Comment by Lisias** [comment]
So, TL;DR: after a whole day screwing up with the `KSPe.Light.TweakScale`, I realised that I was chasing my tail.

**Right now** I have a 2.4.6.21 TweakScale compiled against `KSPe` full borking on KSP 1.4.3 **with or without** the presence of the `KSPe.Light` on the system - being it the good one, the bad one, the mangled one, whatever. This DLL was never the source of the problem, as it appears.

Screenshot of the problem after loading the craft file I posted above with the TweakScale config section mangled to simulate a migration from a `free` scaletype into a `stack` one:

![screenshot89](https://user-images.githubusercontent.com/64334/216520703-14330f72-c53f-4f28-a5ab-bf9b1990f094.png)

The strutCube gets so big that it's outside the VAB, and all we can see from inside it is its shadows. Or, at least, it's what I understood from the picture.

**Comment by Lisias** [comment]
Well, this is definitively not a `KSPe.Light` problem. The almost infinite (and unfortunate) series os testings leaded up to a situation where the problem **is always happening**.

So I now have three possibilities:

1. A very subtle bork on KSPe itself, perhaps a static variable that is failing to be initialised on some circunstances.
2. The same, but on TweakScale
3. A new happening of a very, very [weird problem that once happened on Module Manager](https://forum.kerbalspaceprogram.com/index.php?/topic/50533-18x-112x-module-manager-422-june-18th-2022-the-heatwave-edition/page/288/#comment-3945798) some years ago with me and with FreeThinker too - and that was solved by… **a reboot**.

**Comment by Lisias** [comment]
I rebooted the machine. Nothing changed.

I'm considering that I had interpreted this last error wrongly. It doesn't looks exactly as the original problem, it's a bit different - I may had trying to get rid of a new problem while thinking it's the original one.

I will need to redo the initial steps on a clean 1.4.3 installment, with clean binaries *et all*.

But this fight will be fought another day.

**Comment by Lisias** [comment]
**HA!!**

I finally reproduced the **second** misbehaviour, and yeah, it's an issue on TweakScale.

This one is on TS:

![](https://user-images.githubusercontent.com/64334/216520703-14330f72-c53f-4f28-a5ab-bf9b1990f094.png)

and is going to be fixed there.

The first misbehaviour, however, are still work in progress here. The mishap on TS's code **does not** explains why the presence of an alien DLL (assuming it's not a coincidente) was triggering the issue reported on the OP.

**Comment by Lisias** [comment]
This last misbehaviour was tackled down on issue https://github.com/net-lisias-ksp/TweakScale/issues/287 and… It was not exactly a TweakScale bug.

See the issue for detais, but TL;DR: something mysteriously changed on how `PartModule.OnLoad(ConfigNode)` handles the ConfigNode contents, and this change royally screwed me on the ScaleType Migration Code.

**Comment by Lisias** [comment]
AAAAND… Right now, TweakScale 2.4.6.19 is not borking the same as it was when I opened this issue.

All I get is the (now expected) problem described on this [post](https://github.com/net-lisias-ksp/KSPe/issues/46#issuecomment-1416840785) (the shadows, not the big truss)

The problem 1 just vanished from the rig. Tested it on 1.4.3 and on 1.12.5, exactly as I did 2 days ago. Now I only get the problem 2, that it's really **another** alien problem because mangling the ConfigNode **worked on March 2022** when I first published it.

Hell. This is not a KSPe (Light or not) issue neither. It's like TweakScale was (is) being targeted, so this issue is going back to TweakScale.
